src/Message.java

```diff
@@ -0,0 +1,114 @@
+import java.io.Serializable;
+import java.util.ArrayList; 
+
+public class Message implements Serializable {
+		
+	protected MessageType type;
+	protected String[] users;
+	protected String username;
+	protected String password;
+	protected String login_status;
+	protected String room_id;
+	protected String user_id;
+	protected String contents;
+	protected ArrayList<Room> rooms; 
+
+	//TODO: no private setType() like assignment 5?
```

Comment:
we can consider `type` to be immutable since it's only set in the constructor so a `setType()` method isn't necessary